apps/studio/src/server/modules/collection/collection.router.ts

```diff
@@ -353,20 +353,30 @@ export const collectionRouter = router({
     ),
   getCollections: protectedProcedure
     .input(getCollectionsSchema)
-    .query(async ({ ctx, input: { siteId } }) => {
+    .query(async ({ ctx, input: { siteId, hasChildren } }) => {
       // will need permissions to fetch all collections for a site
       await validateUserPermissionsForSite({
         siteId,
         action: "read",
         userId: ctx.user.id,
       })
 
-      return db
-        .selectFrom("Resource")
+      let query = db.selectFrom("Resource")
+
+      if (hasChildren) {
+        query = query.innerJoin(
+          "Resource as children",
+          "Resource.id",
+          "children.parentId",
+        )
+      }
```

Comment:
collections are allowed to have pages, links and collection meta under them. this assumes that all these resource types are valid children for collections. i'm assuming this is ok? (though we need to be aware that if we have only collection meta under a collection, the collection itself will be empty)